src/work/SimpleDAO.sol

```diff
@@ -111,75 +131,156 @@ contract SimpleDAO {
         yesVoteThreshold = _yesVoteThreshold;
     }
 
+    /**
+     * @dev Allows a member to delegate their voting power to another member.
+     * @param _delegate The address of the member to delegate voting power to.
+     */
+    function delegateVotingPower(address _delegate) external onlyMember {
+        require(members[_delegate].isMember, "Delegate must be a member");
+        members[msg.sender].delegate = _delegate;
+    }
+
+    /**
+     * @dev Allows a member to revoke their voting power delegation.
+     */
+    function revokeDelegation() external onlyMember {
+        delete members[msg.sender].delegate;
+    }
+
     /**
      * @dev Allows a member to create a new proposal.
+     * @param _proposalType The type of the proposal.
      * @param _description The description of the proposal.
      * @param _expiryTime The expiry time of the proposal.
      * @param _target The target address for the proposal execution.
      * @param _callData The call data for the proposal execution.
      */
-    function createProposal(string memory _description, uint _expiryTime, address _target, bytes memory _callData) external onlyMember {
+    function createProposal(
+        ProposalType _proposalType,
+        string memory _description,
+        uint _expiryTime,
+        address _target,
+        bytes memory _callData
+    ) external payable onlyMember notEmergencyStop {
         require(bytes(_description).length <= MAX_DESCRIPTION_LENGTH, "Description too long");
         require(_expiryTime > block.timestamp, "Expiry time must be in the future");
+        require(msg.value == proposalDeposit, "Incorrect proposal deposit");
 
         Proposal storage newProposal = proposals.push();
+        newProposal.proposalType = _proposalType;
+        newProposal.proposalState = ProposalState.Pending;
         newProposal.description = _description;
         newProposal.creationTime = block.timestamp;
         newProposal.expiryTime = _expiryTime;
         newProposal.target = _target;
         newProposal.callData = _callData;
+        newProposal.deposit = msg.value;
 
-        emit ProposalCreated(proposals.length - 1, _description, _expiryTime, _target, _callData);
+        emit ProposalCreated(proposals.length - 1, _proposalType, _description, _expiryTime, _target, _callData);
     }
 
     /**
      * @dev Allows a member to vote on a proposal.
      * @param _proposalId The ID of the proposal to vote on.
      * @param _vote The vote (true for yes, false for no).
      */
-    function vote(uint _proposalId, bool _vote) external onlyMember {
+    function vote(uint _proposalId, bool _vote) external onlyMember notEmergencyStop {
         Proposal storage proposal = proposals[_proposalId];
+        require(proposal.proposalState == ProposalState.Active, "Proposal is not active");
```

Comment:
you need an activeProposal method to change the state of the proposal from pending to active